packages/protocol/test/ACE/validators/privateRange/index.js

```diff
@@ -0,0 +1,324 @@
+            proof.challenge = proof.challengeHash.redKeccak();
+            const data = proof.encodeABI();
+            await truffleAssert.reverts(
+                privateRangeValidator.validatePrivateRange(data, sender, bn128.CRS),
+                truffleAssert.ErrorType.REVERT,
+            );
+        });
+
+        it('should fail if publicValue NOT integrated into challenge variable', async () => {
+            const { originalNote, comparisonNote, utilityNote } = await getDefaultNotes();
+            const proof = new PrivateRangeProof(originalNote, comparisonNote, utilityNote, sender);
+            // Second element should have been the publicValue
+            proof.challengeHash = new Keccak();
+            proof.constructChallengeRecurse([proof.sender, proof.publicOwner, proof.notes, proof.blindingFactors]);
+            proof.challenge = proof.challengeHash.redKeccak();
+            const data = proof.encodeABI();
+            await truffleAssert.reverts(
+                privateRangeValidator.validatePrivateRange(data, sender, bn128.CRS),
+                truffleAssert.ErrorType.REVERT,
+            );
+        });
+
+        it('should fail if publicOwner address NOT integrated into challenge variable', async () => {
+            const { originalNote, comparisonNote, utilityNote } = await getDefaultNotes();
+            const proof = new PrivateRangeProof(originalNote, comparisonNote, utilityNote, sender);
+            // Third element should have been the publicOwner
+            proof.challengeHash = new Keccak();
+            proof.constructChallengeRecurse([proof.sender, proof.publicValue, proof.notes, proof.blindingFactors]);
+            proof.challenge = proof.challengeHash.redKeccak();
+            const data = proof.encodeABI();
+            await truffleAssert.reverts(
+                privateRangeValidator.validatePrivateRange(data, sender, bn128.CRS),
+                truffleAssert.ErrorType.REVERT,
+            );
+        });
+
+        it('should fail if notes NOT integrated into challenge variable', async () => {
+            const { originalNote, comparisonNote, utilityNote } = await getDefaultNotes();
+            const proof = new PrivateRangeProof(originalNote, comparisonNote, utilityNote, sender);
+            // Fourth element should have been the notes
+            proof.challengeHash = new Keccak();
+            proof.constructChallengeRecurse([proof.sender, proof.publicValue, proof.publicOwner, proof.blindingFactors]);
+            proof.challenge = proof.challengeHash.redKeccak();
+            const data = proof.encodeABI();
+            await truffleAssert.reverts(
+                privateRangeValidator.validatePrivateRange(data, sender, bn128.CRS),
+                truffleAssert.ErrorType.REVERT,
+            );
+        });
+
+        it('should fail if blindingFactors NOT integrated into challenge variable', async () => {
```

Comment:
I think the following test would be useful to add, to check that we're validating our bilinear pairing correctly:

Create a situation where the input/output/utility note values satisfy the private range proof, but the utility note is generated over a different setup key than the other notes (i.e. `y` is not the same, so the bilinear pairing test should fail). 

We should repeat the above test for permutations of all 3 notes (so one where the input note is over the wrong setup key, and one for the output note)

Reply:
Very good point. As discussed, we want to implement this across all proofs. Will add the task to our scheduling tool and break it out seperately from this PR